plugin.video.retrospect/resources/libs/envcontroller.py

```diff
@@ -0,0 +1,433 @@
+        with the amount of seconds as defined in cacheTime.
+
+        Expired items are deleted.
+        :param str path:        The cache path to clean.
+        :param int cache_time:  The minimum (in seconds) of files that will be deleted.
+        :param str mask:        The file mask to consider when cleaning the cache.
+
+        """
+
+        # let's import htis one here
+        import fnmatch
+
+        try:
+            Logger.info("Cleaning up cache in '%s' that is older than %s days",
+                        os.path.join(path, "**", mask), cache_time / 24 / 3600)
+
+            if not os.path.exists(path):
+                Logger.info("Did not cleanup cache: folder does not exist")
+                return
+
+            delete_count = 0
+            file_count = 0
+
+            #for item in os.listdir(path):
+            current_dir = None
+            for root, dirs, files in os.walk(path):
+                if current_dir != root:
+                    Logger.debug("Cleaning cache folder: %s", root)
+                    current_dir = root
+
+                for basename in files:
+                    if fnmatch.fnmatch(basename, mask):
+                        filename = os.path.join(root, basename)
+                        Logger.trace("Inspecting: %s", filename)
+                        file_count += 1
+                        create_time = os.path.getctime(filename)
+                        if create_time + cache_time < time.time():
+                            os.remove(filename)
+                            Logger.debug("Removed file: %s", filename)
+                            delete_count += 1
+
+            Logger.info("Removed %s of %s files from cache in: '%s'", delete_count, file_count, path)
+        except:
+            Logger.critical("Error cleaning the cachefolder: %s", path, exc_info=True)
+
+    def __get_python_version(self):
+        """Returns the current python version
+
+        Returns:
+        Python version in the #.#.# format
+
+        """
+
+        major = sys.version_info[0]
+        minor = sys.version_info[1]
+        build = sys.version_info[2]
+        return "%s.%s.%s" % (major, minor, build)
+
+    def __get_environment(self):
+        """ Gets the type of environment for Kodi in a string:
+
+        * Linux   - Normal Linux
+        * Linux64 - 64-bit Linux
+        * OS X    - For Apple decices
+        * win32   - Windows / Native Xbox
+
+        :return: String representation for the current environment.
+
+        """
+
+        is_x64 = struct.calcsize("P") == 8
+
+        env = os.environ.get("OS", "win32")
+        if env == "Linux":
+            if is_x64:
+                return "Linux64"
+            return "Linux"
+
+        elif env == "OS X":
+            return "OS X"
+
+        else:
+            if is_x64:
+                return "Win64"
+
+            return "Win32"
+
+    def __is_repo_available(self, config, return_name=False):
```

Comment:
xbmc addon repository prevails above any other. Please drop.

Reply:
No longer needed. Removed it.